### ISSUE TITLE
HA 2024.10.0b0 breaks integration

### ISSUE THREAD

**Issue by mhoogenbosch** [pinkywafer/Anniversaries#165]
Error:

```
Logger: homeassistant.config_entries
Bron: config_entries.py:594
Eerst voorgekomen: 21:30:27 (8 gebeurtenissen)
Laatst gelogd: 21:30:27

Error setting up entry Lieke verjaardag for anniversaries
Error setting up entry Robyn verjaardag for anniversaries
Error setting up entry Jente verjaardag for anniversaries
Error setting up entry Pepijn verjaardag for anniversaries
Error setting up entry Wessel verjaardag for anniversaries
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 594, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/anniversaries/__init__.py", line 62, in async_setup_entry
    config_entry.options = config_entry.data
    ^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 443, in __setattr__
    raise AttributeError(
AttributeError: options cannot be changed directly, use async_update_entry instead
```

Integration version 5.2.0

**Comment by bdraco** [comment]
https://developers.home-assistant.io/blog/2024/02/12/async_update_entry/

**Comment by rononline** [comment]
@pinkywafer is this integration still alive?

**Comment by cloudbr34k84** [comment]
adding a +1 as im seeing this as well

**Comment by rononline** [comment]
I came across this integration: https://github.com/Miicroo/ha-birthdays but I have to try that. If it works the same as this one.

**Comment by cloudbr34k84** [comment]
> I came across this integration: https://github.com/Miicroo/ha-birthdays but I have to try that. If it works the same as this one.

im installing now to try

**Comment by mhoogenbosch** [comment]
> I came across this integration: https://github.com/Miicroo/ha-birthdays but I have to try that. If it works the same as this one.

Thanks! Installed it and it works. For me, the only requirement is to have a countdown day timer to this specific date, and these are only birthdays. So for me this works great! Removed Anniversaries for now.

**Comment by TheFes** [comment]
I see there is is PR linked to fix it, any idea when there will be a release?

**Comment by broyuken** [comment]
This might be dead, maybe move to polos fork?

**Comment by JeffCrum1** [comment]
@pinkywafer

So, is the hassio queen going to do anything here?

**Comment by MdB-Gryphon** [comment]
Same Issue when I upgraded to HA 2024.10.0 yesterday.

**Comment by MdB-Gryphon** [comment]
Change /config/custom_components/anniversaries/__init__.py on line 52 with the code below and the integration will work again

async def async_setup_entry(hass, config_entry):
    """Set up this integration using UI."""
    if config_entry.source == config_entries.SOURCE_IMPORT:
        # Set up using YAML
        hass.async_create_task(hass.config_entries.async_remove(config_entry.entry_id))
        return False

    # Log startup message
    _LOGGER.info(
        CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
    )

    # Update config entry options using async_update_entry
    hass.config_entries.async_update_entry(
        config_entry, options=config_entry.data
    )

    # Add update listener
    config_entry.add_update_listener(update_listener)

    # Forward setup to platform (e.g., sensor)
    hass.async_add_job(
        hass.config_entries.async_forward_entry_setup(config_entry, PLATFORM)
    )

    return True

**Comment by dengi76** [comment]
> Измените /config/custom_components/anniversaries/ **init** .py в строке 52 на код ниже, и интеграция снова заработает.
> 
> async def async_setup_entry(hass, config_entry): """Настройте эту интеграцию с помощью пользовательского интерфейса.""" if config_entry.source == config_entries.SOURCE_IMPORT: # Настройте с помощью YAML hass.async_create_task(hass.config_entries.async_remove(config_entry.entry_id)) return False
> 
> ```
> # Log startup message
> _LOGGER.info(
>     CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
> )
> 
> # Update config entry options using async_update_entry
> hass.config_entries.async_update_entry(
>     config_entry, options=config_entry.data
> )
> 
> # Add update listener
> config_entry.add_update_listener(update_listener)
> 
> # Forward setup to platform (e.g., sensor)
> hass.async_add_job(
>     hass.config_entries.async_forward_entry_setup(config_entry, PLATFORM)
> )
> 
> return True
> ```

Thank you, it helped!

**Comment by clefus** [comment]
+1 on this issue.  Any plans for calendar to be fixed?  Prefer it over birthdays from what I have seen.

**Comment by broyuken** [comment]
just go in and make the fix above, works fine again.

**Comment by BebeMischa** [comment]
> @pinkywafer
> 
> So, is the hassio queen going to do anything here?

Looks like the Queen left the palace...

**Comment by revoltadodan** [comment]
@ChirpyTurnip
Delete all the contents of the init.py file and paste the following code:

```
"""The Anniversaries Integration"""
import logging
from homeassistant import config_entries
from homeassistant.helpers import discovery

from integrationhelper.const import CC_STARTUP_VERSION

from .const import (
    CONF_SENSORS,
    CONF_DATE_TEMPLATE,
    DOMAIN,
    ISSUE_URL,
    PLATFORM,
    VERSION,
    CONFIG_SCHEMA,
)

_LOGGER = logging.getLogger(__name__)

async def async_setup(hass, config):
    """Set up this component using YAML."""
    if config.get(DOMAIN) is None:
        # config flow setup
        return True

    # Log startup message
    _LOGGER.info(
        CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
    )
    
    platform_config = config[DOMAIN].get(CONF_SENSORS, {})

    # If platform is not enabled, skip.
    if not platform_config:
        return False

    for entry in platform_config:
        hass.async_create_task(
            discovery.async_load_platform(hass, PLATFORM, DOMAIN, entry, config)
        )
    hass.async_create_task(
        hass.config_entries.flow.async_init(
            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
        )
    )
    return True

async def async_setup_entry(hass, config_entry):
    """Set up this integration using UI."""
    if config_entry.source == config_entries.SOURCE_IMPORT:
        # Set up using YAML
        hass.async_create_task(hass.config_entries.async_remove(config_entry.entry_id))
        return False

    # Log startup message
    _LOGGER.info(
        CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
    )

    # Update config entry options using async_update_entry
    hass.config_entries.async_update_entry(
        config_entry, options=config_entry.data
    )

    # Add update listener
    config_entry.add_update_listener(update_listener)

    # Forward setup to platform (e.g., sensor)
    hass.async_add_job(
        hass.config_entries.async_forward_entry_setup(config_entry, PLATFORM)
    )

    return True

async def async_remove_entry(hass, config_entry):
    """Handle removal of an entry."""
    try:
        await hass.config_entries.async_forward_entry_unload(config_entry, PLATFORM)
        _LOGGER.info(
            "Successfully removed sensor from the Anniversaries integration"
        )
    except ValueError:
        pass

async def update_listener(hass, entry):
    """Update listener."""
    entry.data = entry.options
    await hass.config_entries.async_forward_entry_unload(entry, PLATFORM)
    hass.async_add_job(hass.config_entries.async_forward_entry_setup(entry, PLATFORM))
```

**Comment by ChirpyTurnip** [comment]
That fixed it!  :-)

**Comment by clefus** [comment]
Fixed for me as well.  Thank you!

**Comment by BebeMischa** [comment]
Thank you very much, @revoltadodan ;-)

**Comment by babygiraffe05** [comment]
That fixed it for me as well... however, note that it required a FULL RESTART of HA. Simply reloading yaml does not fix the issue. Be sure you select full restart.

**Comment by JeffCrum1** [comment]
Thanks @MdB-Gryphon and @revoltadodan.

Y'all are awesome!!!

**Comment by mabusdogma** [comment]
Thanks! Now it works, what I did was

1.  Copy the lines described above:

> ```
> """The Anniversaries Integration"""
> import logging
 . . .
> hass.async_add_job(hass.config_entries.async_forward_entry_setup(entry, PLATFORM))
> ```

2. Go to File Editor (install the addon if you don't have it), enter in "custom components" folder and then in "anniversaries" folder
3. Open init.py, and replace what it has with the copied lines from above
4. Restart home assistant

Thanks

**Comment by juanmijm** [comment]
Thanks Revoltadodan, works great!

**Comment by MdB-Gryphon** [comment]
I made an modification to the code, so that it uses hass.config_entries.async_forward_entry_setups method and replaces hass.config_entries.async_forward_entry_setup. That removes the error in the log that it is deprecated and will be removed in Home Assistant 2025.6. See: https://developers.home-assistant.io/blog/2024/06/12/async_forward_entry_setups



"""The Anniversaries Integration"""
import logging
from homeassistant import config_entries
from homeassistant.helpers import discovery

from integrationhelper.const import CC_STARTUP_VERSION

from .const import (
    CONF_SENSORS,
    CONF_DATE_TEMPLATE,
    DOMAIN,
    ISSUE_URL,
    PLATFORM,
    VERSION,
    CONFIG_SCHEMA,
)

_LOGGER = logging.getLogger(__name__)

async def async_setup(hass, config):
    """Set up this component using YAML."""
    if config.get(DOMAIN) is None:
        # Config flow setup if no YAML config exists
        return True

    # Log startup message
    _LOGGER.info(
        CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
    )

    platform_config = config[DOMAIN].get(CONF_SENSORS, {})

    # If no platform is enabled, skip setup
    if not platform_config:
        return False

    # Load platform configuration for each entry
    for entry in platform_config:
        hass.async_create_task(
            discovery.async_load_platform(hass, PLATFORM, DOMAIN, entry, config)
        )

    # Initiate config flow to import YAML config
    hass.async_create_task(
        hass.config_entries.flow.async_init(
            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
        )
    )

    return True

async def async_setup_entry(hass, config_entry):
    """Set up this integration using UI."""
    if config_entry.source == config_entries.SOURCE_IMPORT:
        # Remove UI config entry if set up via YAML
        await hass.config_entries.async_remove(config_entry.entry_id)
        return False

    # Log startup message
    _LOGGER.info(
        CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
    )

    # Safely update entry options if needed
    hass.config_entries.async_update_entry(
        config_entry, options=config_entry.data
    )

    # Add update listener for configuration changes
    config_entry.add_update_listener(update_listener)

    # Set up the platforms using the new `async_forward_entry_setups`
    await hass.config_entries.async_forward_entry_setups(config_entry, [PLATFORM])

    return True

async def async_unload_entry(hass, config_entry):
    """Unload a config entry."""
    # Unload the platform using the new `async_forward_entry_unload`
    if await hass.config_entries.async_forward_entry_unload(config_entry, [PLATFORM]):
        _LOGGER.info(f"Successfully unloaded {PLATFORM} for {DOMAIN}")
        return True
    else:
        _LOGGER.error(f"Error unloading {PLATFORM} for {DOMAIN}")
        return False

async def async_remove_entry(hass, config_entry):
    """Handle removal of a config entry."""
    # Ensure the platform is unloaded before removing the entry
    await async_unload_entry(hass, config_entry)
    _LOGGER.info(f"Successfully removed entry for {DOMAIN}")

async def update_listener(hass, entry):
    """Handle updates to the config entry."""
    # Update the entry's data based on options
    hass.config_entries.async_update_entry(entry, data=entry.options)

    # Unload and reload platform to apply new settings
    await async_unload_entry(hass, entry)
    await async_setup_entry(hass, entry)

**Comment by MdB-Gryphon** [comment]
I've made an adjustment so the error hass.config_entries.async_forward_entry_setup is deprecated and will be removed in Home Assistant 2025.6. is also gone

See https://developers.home-assistant.io/blog/2024/06/12/async_forward_entry_setups/


"""The Anniversaries Integration"""
import logging
from homeassistant import config_entries
from homeassistant.helpers import discovery

from integrationhelper.const import CC_STARTUP_VERSION

from .const import (
    CONF_SENSORS,
    CONF_DATE_TEMPLATE,
    DOMAIN,
    ISSUE_URL,
    PLATFORM,
    VERSION,
    CONFIG_SCHEMA,
)

_LOGGER = logging.getLogger(__name__)

async def async_setup(hass, config):
    """Set up this component using YAML."""
    if config.get(DOMAIN) is None:
        # Config flow setup if no YAML config exists
        return True

    # Log startup message
    _LOGGER.info(
        CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
    )

    platform_config = config[DOMAIN].get(CONF_SENSORS, {})

    # If no platform is enabled, skip setup
    if not platform_config:
        return False

    # Load platform configuration for each entry
    for entry in platform_config:
        hass.async_create_task(
            discovery.async_load_platform(hass, PLATFORM, DOMAIN, entry, config)
        )

    # Initiate config flow to import YAML config
    hass.async_create_task(
        hass.config_entries.flow.async_init(
            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
        )
    )

    return True

async def async_setup_entry(hass, config_entry):
    """Set up this integration using UI."""
    if config_entry.source == config_entries.SOURCE_IMPORT:
        # Remove UI config entry if set up via YAML
        await hass.config_entries.async_remove(config_entry.entry_id)
        return False

    # Log startup message
    _LOGGER.info(
        CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
    )

    # Safely update entry options if needed
    hass.config_entries.async_update_entry(
        config_entry, options=config_entry.data
    )

    # Add update listener for configuration changes
    config_entry.add_update_listener(update_listener)

    # Set up the platforms using the new `async_forward_entry_setups`
    await hass.config_entries.async_forward_entry_setups(config_entry, [PLATFORM])

    return True

async def async_unload_entry(hass, config_entry):
    """Unload a config entry."""
    # Unload the platform using the new `async_forward_entry_unload`
    if await hass.config_entries.async_forward_entry_unload(config_entry, [PLATFORM]):
        _LOGGER.info(f"Successfully unloaded {PLATFORM} for {DOMAIN}")
        return True
    else:
        _LOGGER.error(f"Error unloading {PLATFORM} for {DOMAIN}")
        return False

async def async_remove_entry(hass, config_entry):
    """Handle removal of a config entry."""
    # Ensure the platform is unloaded before removing the entry
    await async_unload_entry(hass, config_entry)
    _LOGGER.info(f"Successfully removed entry for {DOMAIN}")

async def update_listener(hass, entry):
    """Handle updates to the config entry."""
    # Update the entry's data based on options
    hass.config_entries.async_update_entry(entry, data=entry.options)

    # Unload and reload platform to apply new settings
    await async_unload_entry(hass, entry)
    await async_setup_entry(hass, entry)

**Comment by juanmijm** [comment]
> I've made an adjustment so the error hass.config_entries.async_forward_entry_setup is deprecated and will be removed in Home Assistant 2025.6. is also gone
> 
> See https://developers.home-assistant.io/blog/2024/06/12/async_forward_entry_setups/
> 
> """The Anniversaries Integration""" import logging from homeassistant import config_entries from homeassistant.helpers import discovery
> 
> from integrationhelper.const import CC_STARTUP_VERSION
> 
> from .const import ( CONF_SENSORS, CONF_DATE_TEMPLATE, DOMAIN, ISSUE_URL, PLATFORM, VERSION, CONFIG_SCHEMA, )
> 
> _LOGGER = logging.getLogger(**name**)
> 
> async def async_setup(hass, config): """Set up this component using YAML.""" if config.get(DOMAIN) is None: # Config flow setup if no YAML config exists return True
> 
> ```
> # Log startup message
> _LOGGER.info(
>     CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
> )
> 
> platform_config = config[DOMAIN].get(CONF_SENSORS, {})
> 
> # If no platform is enabled, skip setup
> if not platform_config:
>     return False
> 
> # Load platform configuration for each entry
> for entry in platform_config:
>     hass.async_create_task(
>         discovery.async_load_platform(hass, PLATFORM, DOMAIN, entry, config)
>     )
> 
> # Initiate config flow to import YAML config
> hass.async_create_task(
>     hass.config_entries.flow.async_init(
>         DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
>     )
> )
> 
> return True
> ```
> 
> async def async_setup_entry(hass, config_entry): """Set up this integration using UI.""" if config_entry.source == config_entries.SOURCE_IMPORT: # Remove UI config entry if set up via YAML await hass.config_entries.async_remove(config_entry.entry_id) return False
> 
> ```
> # Log startup message
> _LOGGER.info(
>     CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
> )
> 
> # Safely update entry options if needed
> hass.config_entries.async_update_entry(
>     config_entry, options=config_entry.data
> )
> 
> # Add update listener for configuration changes
> config_entry.add_update_listener(update_listener)
> 
> # Set up the platforms using the new `async_forward_entry_setups`
> await hass.config_entries.async_forward_entry_setups(config_entry, [PLATFORM])
> 
> return True
> ```
> 
> async def async_unload_entry(hass, config_entry): """Unload a config entry.""" # Unload the platform using the new `async_forward_entry_unload` if await hass.config_entries.async_forward_entry_unload(config_entry, [PLATFORM]): _LOGGER.info(f"Successfully unloaded {PLATFORM} for {DOMAIN}") return True else: _LOGGER.error(f"Error unloading {PLATFORM} for {DOMAIN}") return False
> 
> async def async_remove_entry(hass, config_entry): """Handle removal of a config entry.""" # Ensure the platform is unloaded before removing the entry await async_unload_entry(hass, config_entry) _LOGGER.info(f"Successfully removed entry for {DOMAIN}")
> 
> async def update_listener(hass, entry): """Handle updates to the config entry.""" # Update the entry's data based on options hass.config_entries.async_update_entry(entry, data=entry.options)
> 
> ```
> # Unload and reload platform to apply new settings
> await async_unload_entry(hass, entry)
> await async_setup_entry(hass, entry)
> ```

please can you paste entire .py file better? Thanks

**Comment by JeffCrum1** [comment]
@juanmijm the entire file was posted and copyable above.

**Comment by juanmijm** [comment]
> @juanmijm the entire file was posted and copyable above.

but the entire file above does not have the modifications that MdB-Gryphon mentions.

**Comment by broyuken** [comment]
> I've made an adjustment so the error hass.config_entries.async_forward_entry_setup is deprecated and will be removed in Home Assistant 2025.6. is also gone
> 
> See https://developers.home-assistant.io/blog/2024/06/12/async_forward_entry_setups/
> 
> """The Anniversaries Integration""" import logging from homeassistant import config_entries from homeassistant.helpers import discovery
> 
> from integrationhelper.const import CC_STARTUP_VERSION
> 
> from .const import ( CONF_SENSORS, CONF_DATE_TEMPLATE, DOMAIN, ISSUE_URL, PLATFORM, VERSION, CONFIG_SCHEMA, )
> 
> _LOGGER = logging.getLogger(**name**)
> 
> async def async_setup(hass, config): """Set up this component using YAML.""" if config.get(DOMAIN) is None: # Config flow setup if no YAML config exists return True
> 
> ```
> # Log startup message
> _LOGGER.info(
>     CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
> )
> 
> platform_config = config[DOMAIN].get(CONF_SENSORS, {})
> 
> # If no platform is enabled, skip setup
> if not platform_config:
>     return False
> 
> # Load platform configuration for each entry
> for entry in platform_config:
>     hass.async_create_task(
>         discovery.async_load_platform(hass, PLATFORM, DOMAIN, entry, config)
>     )
> 
> # Initiate config flow to import YAML config
> hass.async_create_task(
>     hass.config_entries.flow.async_init(
>         DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
>     )
> )
> 
> return True
> ```
> 
> async def async_setup_entry(hass, config_entry): """Set up this integration using UI.""" if config_entry.source == config_entries.SOURCE_IMPORT: # Remove UI config entry if set up via YAML await hass.config_entries.async_remove(config_entry.entry_id) return False
> 
> ```
> # Log startup message
> _LOGGER.info(
>     CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
> )
> 
> # Safely update entry options if needed
> hass.config_entries.async_update_entry(
>     config_entry, options=config_entry.data
> )
> 
> # Add update listener for configuration changes
> config_entry.add_update_listener(update_listener)
> 
> # Set up the platforms using the new `async_forward_entry_setups`
> await hass.config_entries.async_forward_entry_setups(config_entry, [PLATFORM])
> 
> return True
> ```
> 
> async def async_unload_entry(hass, config_entry): """Unload a config entry.""" # Unload the platform using the new `async_forward_entry_unload` if await hass.config_entries.async_forward_entry_unload(config_entry, [PLATFORM]): _LOGGER.info(f"Successfully unloaded {PLATFORM} for {DOMAIN}") return True else: _LOGGER.error(f"Error unloading {PLATFORM} for {DOMAIN}") return False
> 
> async def async_remove_entry(hass, config_entry): """Handle removal of a config entry.""" # Ensure the platform is unloaded before removing the entry await async_unload_entry(hass, config_entry) _LOGGER.info(f"Successfully removed entry for {DOMAIN}")
> 
> async def update_listener(hass, entry): """Handle updates to the config entry.""" # Update the entry's data based on options hass.config_entries.async_update_entry(entry, data=entry.options)
> 
> ```
> # Unload and reload platform to apply new settings
> await async_unload_entry(hass, entry)
> await async_setup_entry(hass, entry)
> ```

Could you please just link your repo and/or do a PR for this instead of just having it as a comment?

**Comment by gizmoha** [comment]
das kopieren der vollständigen codes hat bei mir auch geholfen. vielen dank. ihr seid spitze

**Comment by homefuzz-oz** [comment]
Brilliant, @DanJr90. Thank you!

**Comment by toofewacres** [comment]
Init file replacement worked perfect. Thanks

**Comment by GitHubxtsnl** [comment]
> @ChirpyTurnip Delete all the contents of the init.py file and paste the following code:
> 
> ```
> """The Anniversaries Integration"""
> 
> ..................
> 
>  
>     hass.async_add_job(hass.config_entries.async_forward_entry_setup(entry, PLATFORM))
> ```

this worked for me, thanks.

**Comment by juanmijm** [comment]
@DanJr90's solution works for now, but it gives a warning that it uses components that will be deprecated, you have to implement the solution proposed by @MdB-Gryphon, but unfortunately we don't have the complete code and when pasting the fragments the indentations fail, also there are parts of the code that don't match those of @DanJr90, so I don't know how to completely solve it (I'm a newbie at this), so it would be great if @MdB-Gryphon could put up the complete code, thanks.

**Comment by PHeonix25** [comment]
Just to combine the two answers above, here's the full code snippet that doesn't error and doesn't throw warnings. 
Thanks to @MdB-Gryphon and @DanJr90!

```python
"""The Anniversaries Integration"""
import logging
from homeassistant import config_entries
from homeassistant.helpers import discovery
from integrationhelper.const import CC_STARTUP_VERSION

from .const import (
    CONF_SENSORS,
    DOMAIN,
    ISSUE_URL,
    PLATFORM,
    VERSION,
)

_LOGGER = logging.getLogger(__name__)

async def async_setup(hass, config):
    """Set up this component using YAML."""
    if config.get(DOMAIN) is None:
        # Config flow setup if no YAML config exists
        return True
    
    # Log startup message
    _LOGGER.info(
        CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
    )
    platform_config = config[DOMAIN].get(CONF_SENSORS, {})
    
    # If no platform is enabled, skip setup
    if not platform_config:
        return False
    
    # Load platform configuration for each entry
    for entry in platform_config:
        hass.async_create_task(
            discovery.async_load_platform(hass, PLATFORM, DOMAIN, entry, config)
        )
    
    # Initiate config flow to import YAML config
    hass.async_create_task(
        hass.config_entries.flow.async_init(
            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
        )
    )

    return True

async def async_setup_entry(hass, config_entry):
    """Set up this integration using UI."""
    if config_entry.source == config_entries.SOURCE_IMPORT:
        # Remove UI config entry if set up via YAML
        await hass.config_entries.async_remove(config_entry.entry_id)
        return False

    # Log startup message
    _LOGGER.info(
        CC_STARTUP_VERSION.format(name=DOMAIN, version=VERSION, issue_link=ISSUE_URL)
    )

    # Safely update entry options if needed
    hass.config_entries.async_update_entry(
        config_entry, options=config_entry.data
    )

    # Add update listener for configuration changes
    config_entry.add_update_listener(update_listener)

    # Set up the platforms using the new `async_forward_entry_setups`
    await hass.config_entries.async_forward_entry_setups(config_entry, [PLATFORM])

    return True

async def async_unload_entry(hass, config_entry):
    """Unload a config entry."""
    # Unload the platform using the new async_forward_entry_unload
    if await hass.config_entries.async_forward_entry_unload(config_entry, [PLATFORM]):
        _LOGGER.info(f"Successfully unloaded {PLATFORM} for {DOMAIN}")
        return True
    else:
        _LOGGER.error(f"Error unloading {PLATFORM} for {DOMAIN}")
        return False

async def async_remove_entry(hass, config_entry):
    """Handle removal of a config entry."""
    # Ensure the platform is unloaded before removing the entry
    await async_unload_entry(hass, config_entry)
    _LOGGER.info(f"Successfully removed entry for {DOMAIN}")

async def update_listener(hass, entry):
    """Handle updates to the config entry."""
    # Update the entry's data based on options
    hass.config_entries.async_update_entry(entry, data=entry.options)

    # Unload and reload platform to apply new settings
    await async_unload_entry(hass, entry)
    await async_setup_entry(hass, entry)
```

Replace your whole `/custom_components/anniversaries/__init.py__` with the above code block ☝️

**Comment by broyuken** [comment]
Someone please just make this a PR. It’s not a good idea to copy random code from a comment into your systems, and it will get lost in the mix. If you make a PR it can get merged and everyone can use this going forward.

**Comment by PHeonix25** [comment]
Agreed, but unfortunately, there's already multiple PRs open containing subsets of this code and no-one to merge them 😢 
I figured posting this here would lead someone to the `Issues` and now they have a single, simple answer 😁

**Comment by broyuken** [comment]
Well if this component isn’t being merged and is no longer maintained then maybe we can start using your repo instead. It’s still 1000x better than using code from a comment.

**Comment by MdB-Gryphon** [comment]
> Someone please just make this a PR. It’s not a good idea to copy random code from a comment into your systems, and it will get lost in the mix. If you make a PR it can get merged and everyone can use this going forward.

You can find the file here: https://github.com/MdB-Gryphon/Anniversaries/blob/main/__init__.py

**Comment by Stoony1** [comment]
@DanJr90 Thx your FIX its Work nice.

**Comment by pinkywafer** [comment]
closed by #166

**Comment by pinkywafer** [comment]
partially fixed by #166

**Comment by pinkywafer** [comment]
closed by #168

**Comment by JeffCrum1** [comment]
Thanks @pinkywafer and everyone else that worked on this!

It is appreciated!
